frontends/web/src/components/language/language.tsx

```diff
@@ -14,34 +15,60 @@
  * limitations under the License.
  */
 
+import { Component, h, RenderableProps } from 'preact';
 import { /* i18nEditorActive, */ extraLanguages } from '../../i18n/i18n';
-import { Component, h } from 'preact';
-import { translate } from 'react-i18next';
+
+import { translate as translateDecorator, TranslateProps } from '../../decorators/translate';
 import { Dialog } from '../dialog/dialog';
 import * as style from './language.css';
 
-@translate()
-export default class LanguageSwitcher extends Component {
+type TActiveLanguageCodes = 'bg' | 'de' | 'en' | 'es'
+    | 'fr' | 'hi' | 'it' | 'ja' | 'ms' | 'pt'
+    | 'ru' | 'sl' | 'tr' | 'zh';
+
+type TLanguage = {
+    code: TActiveLanguageCodes | string;
```

Comment:
If code can also be a string, then what's the point of defining `TActiveLanguageCodes`. Can the `code` get assigned a random value somehow, that is not among the `TActiveLanguageCodes`?

Reply:
Will this work:

```ts
code: TActiveLanguageCodes
```

(no `| string`)

Reply:
Typescript will error in the tests

![Screen Shot 2021-04-27 at 5 34 19 PM](https://user-images.githubusercontent.com/546900/116269840-d4a17100-a77e-11eb-9363-31f4a6a600eb.png)

https://github.com/digitalbitbox/bitbox-wallet-app/blob/0b03df07ea60ba491bd5eab52b1b43596c09284b/frontends/web/test/components/language/language.test.tsx#L31

But we could just ignore in the tests
```diff
diff --git a/frontends/web/src/components/language/language.tsx b/frontends/web/src/components/language/language.tsx
index 9335e3b7..b7e760fa 100644
--- a/frontends/web/src/components/language/language.tsx
+++ b/frontends/web/src/components/language/language.tsx
@@ -27,7 +27,7 @@ type TActiveLanguageCodes = 'bg' | 'de' | 'en' | 'es'
     | 'ru' | 'sl' | 'tr' | 'zh';
 
 type TLanguage = {
-    code: TActiveLanguageCodes | string;
+    code: TActiveLanguageCodes;
     display: string;
 };
 
diff --git a/frontends/web/test/components/language/language.test.tsx b/frontends/web/test/components/language/language.test.tsx
index 02d0245a..239c3637 100644
--- a/frontends/web/test/components/language/language.test.tsx
+++ b/frontends/web/test/components/language/language.test.tsx
@@ -36,6 +36,7 @@ describe('components/language/language', () => {
             it(`returns exact match (${lang.code})`, (done) => {
                 i18n.changeLanguage(lang.code, (err) => {
                     expect(err).toBe(null);
+                    /* @ts-ignore */
                     const ctx = deep(<LanguageSwitch i18n={i18n} languages={supportedLangs} />);
                     expect(ctx.state('selectedIndex')).toEqual(idx);
                     done();
@@ -46,6 +47,7 @@ describe('components/language/language', () => {
         it('matches main language tag', (done) => {
             i18n.changeLanguage('pt_BR', (err) => {
                 expect(err).toBe(null);
+                /* @ts-ignore */
                 const ctx = deep(<LanguageSwitch i18n={i18n} languages={supportedLangs} />);
                 expect(ctx.state('selectedIndex')).toEqual(2); // 'pt'
                 done();
@@ -55,6 +57,7 @@ describe('components/language/language', () => {
         it('returns default if none matched', (done) => {
             i18n.changeLanguage('it', (err) => {
                 expect(err).toBe(null);
+                /* @ts-ignore */
                 const ctx = deep(<LanguageSwitch i18n={i18n} languages={supportedLangs} />);
                 expect(ctx.state('selectedIndex')).toEqual(0); // 'en'
                 done();
```

Reply:
I think it's perfectly fine to ts-ignore in tests.